File: spring-boot-java-web/src/main/java/com/example/webjsp/mapper/CityMapper.java
package com.example.webjsp.mapper;

import com.example.webjsp.model.City;
import org.apache.ibatis.annotations.Mapper;
import org.apache.ibatis.annotations.Param;

import java.util.HashMap;
import java.util.List;
import java.util.UUID;

@Mapper
public interface CityMapper {
  City selectCity(@Param("id") UUID id);

  List<HashMap<String, Object>> selectAll();

  int selectCount();
}

File: spring-boot-kotlin-web/src/main/kotlin/com/example/webjsp/mapper/CityMapper.kt
package com.example.webjsp.mapper

import com.example.webjsp.model.City
import org.apache.ibatis.annotations.Mapper
import org.apache.ibatis.annotations.Param
import java.util.*
import kotlin.collections.HashMap

@Mapper
interface CityMapper {
  fun selectCity(@Param("id") id: UUID): City
  fun selectAll(): List<HashMap<String, Any>>
  fun selectCount(): Int
}
